test/e2e/provisioner/provisioner_test.go

```diff
@@ -44,19 +46,36 @@ func TestProvisioner(t *testing.T) {
 var _ = BeforeSuite(func() {
 	require.NoError(f.T(), f.Provisioner.EnsureResourcesForInclusterProvisioner())
 
+	params := &contour_api_v1alpha1.ContourDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "projectcontour",
+			Name:      "basic-contour",
+		},
+		Spec: contour_api_v1alpha1.ContourDeploymentSpec{
+			RuntimeSettings: contourDeploymentRuntimeSettings(),
+		},
+	}
+	require.NoError(f.T(), f.Client.Create(context.Background(), params))
+
 	gc := &gatewayapi_v1alpha2.GatewayClass{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "contour",
 		},
 		Spec: gatewayapi_v1alpha2.GatewayClassSpec{
 			ControllerName: gatewayapi_v1alpha2.GatewayController("projectcontour.io/gateway-controller"),
+			ParametersRef: &gatewayapi_v1alpha2.ParametersReference{
+				Group:     "projectcontour.io",
+				Kind:      "ContourDeployment",
+				Namespace: gatewayapi.NamespacePtr(params.Namespace),
+				Name:      params.Name,
+			},
 		},
 	}
```

Comment:
Just to keep coverage on a GatewayClass with no params, I might check the return val of `contourDeploymentRuntimeSettings` first and only create the ContourDeployment / ParametersRef if it's not nil.

Reply:
should be fixed in https://github.com/projectcontour/contour/pull/4529/commits/8d6503914aa6e62b0faf646a9e147fdec6db91d0 and https://github.com/projectcontour/contour/pull/4529/commits/7dc54bc8ad07393195a15e652031eece8c82ea42